Sources/AblyAssetTrackingCore/Configuration/ConnectionConfiguration.swift

```diff
@@ -10,46 +10,92 @@ public enum AuthResult {
     case tokenDetails(TokenDetails)
 }
 
+public class ObjcAuthResultJWT: NSObject {
```

Comment:
All 3 classes below are just wrappers for the underlying field they hold. This requires users to wrap their result in `authCallback` explicitly. However, why not just use String, TokenRequest and TokenDetails directly? They are all unique types.

Reply:
here c98f8a223d3216a189341cb6b06191f73a65f8a5

Reply:
Cool!